aptos-move/framework/aptos-stdlib/sources/cryptography/federated_keyless.move

```diff
@@ -0,0 +1,76 @@
+/// This module implements the Federated Keyless authentication scheme.
+
+module aptos_std::federated_keyless {
+    use aptos_std::bcs_stream;
+    use aptos_std::keyless;
+
+    #[test_only]
+    use aptos_std::bcs;
+    #[test_only]
+    use std::string::{utf8};
+
+    //
+    // Error codes
+    //
+
+    /// There are extra bytes in the input when deserializing a Federated Keyless public key.
+    const E_INVALID_FEDERATED_KEYLESS_PUBLIC_KEY_EXTRA_BYTES: u64 = 1;
+
+    //
+    // Constants
+    //
+
+    //
+    // Structs
+    //
+
+    /// An *unvalidated* any public key: not necessarily an elliptic curve point, just a sequence of 32 bytes
+    struct PublicKey has copy, drop, store {
+        jwk_address: address,
+        keyless_public_key: keyless::PublicKey,
+    }
+
+    //
+    // Functions
+    //
+
+    /// Parses the input bytes into a keyless public key.
+    public fun new_public_key_from_bytes(bytes: vector<u8>): PublicKey {
+        let stream = bcs_stream::new(bytes);
+        let pk = deserialize_public_key(&mut stream);
+        assert!(bcs_stream::has_remaining(&mut stream) == false, std::error::invalid_argument(E_INVALID_FEDERATED_KEYLESS_PUBLIC_KEY_EXTRA_BYTES));
+        pk
+    }
+
+    public fun deserialize_public_key(stream: &mut bcs_stream::BCSStream): PublicKey {
+        let jwk_address = bcs_stream::deserialize_address(stream);
+        let keyless_public_key = keyless::deserialize_public_key(stream);
+        PublicKey { keyless_public_key, jwk_address }
+    }
+
+    public fun new(keyless_public_key: keyless::PublicKey, jwk_address: address): PublicKey {
+        PublicKey { keyless_public_key, jwk_address }
+    }
+
+    #[test]
```

Comment:
For all new unit tests, please create a new file in tests/ (if this dir doesn't exist, create it at the same level as sources and add subdir for cryptography)